### PR TITLE
Integration scripts and models

### PR DESCRIPTION
**Pull request by Jasper-Thanapal**
## What are the changes?
1)Analysed various models on the ppg feature dataset and tested their accuracy. The notebook file containing the analysis is being pushed
2)The best model from the analysis was converted to pickle file and is being pushed.
3)The scripts necessary for extracting sleep metrics from sleep stage data, integrating the model and the extracted features to predict if a person has sleep disorder or not are being pushed.

## Who worked on the changes?
I (Jasper Thanapal) worked on the changes.

### CODE REVIEW

**Comment by AmirZandiehprojects** [comment]
@Jasper-Thanapal I've conducted a security review of your sleep disorder prediction integration.
From a security standpoint, I didn't find any obvious vulnerabilities in the code provided. The implementation appears to follow generally safe practices:

No hardcoded credentials or sensitive information in the scripts.
The code doesn't appear to execute any user-provided input directly.
File operations are limited to reading from predefined CSV files and a model file.
There's no network communication or database operations that could introduce common vulnerabilities.

However, it's worth noting that the security of the overall system also depends on how these scripts are deployed and used in the broader application context.
As a general recommendation, always ensure that input data (CSV files and user inputs) is validated and sanitized before processing, especially when deploying this in a production environment.
Overall, from the code provided, there are no immediate security concerns to address.

**Review by Manan2711 — CHANGES_REQUESTED** [review]
I couldn't find a file that can be used to create or train the model, can you please add that file as well or let me know if it's added somewhere else?

File: data_science/sleep_disorder_prediction/sleep_disorder_scripts/prediction.ipynb [hunk 1]
```diff
@@ -0,0 +1,70 @@
...
+      "REM%: 16.02112676056338\n",
+      "Arousal Index: 4.916201117318436\n"
+     ]
+    }
+   ],
+   "source": [
+    "from predict import SleepMetricsExtended\n",
+    "\n",
+    "# Example usage\n",
+    "sleep_metrics_extended = SleepMetricsExtended(\"samples/sub_10.csv\", \"random_forest_model.pkl\")\n",
```
[the remark this answers is not in the document]
Comment (Jasper-Thanapal) [reply data_science/sleep_disorder_prediction/sleep_disorder_scripts/prediction.ipynb]: Hi Manan, Ive changed the path now to data/sub_10.csv and also uploaded the datasets in the path. Also I have added the file sleep_stage_classifier.ipynb which has the code to train and develop the model.

**Review by Manan2711 — APPROVED** [review]
Everything looks good to be merged. Thank you!

**Pull request merged**